drodrpg/DRODLib/CurrentGame.h

```diff
@@ -399,13 +400,15 @@ class CCurrentGame : public CDbSavedGame
 	CDbLevel *  pLevel;
 	CDbHold *   pHold;
 	CEntranceData *pEntrance;
+	CTotalMapStates *pTotalMapStates;
```

Comment:
I don't think we need to make this var a pointer, calling new and delete on it here and there. Let's simply make it an object, i.e.,
`CTotalMapStates totalMapStates;`
Now you don't need to worry about memory alloc/dealloc and checking whether the pointer is valid, etc.

I'm assuming that Load(...) will always be called -- not only when allocating a new object, for instance.